lesson_07/homework_07.py

```diff
@@ -0,0 +1,209 @@
+# task 1
+""" Задача - надрукувати табличку множення на задане число, але
+лише до максимального значення для добутку - 25.
+Код майже готовий, треба знайти помилки та випраавити\доповнити.
+"""
+
+
+def multiplication_table(number):  # Initialize the appropriate variable
+    multiplier = 1
+    # Complete the while loop condition.
+    while True:
+        result = number * multiplier  # десь тут помилка, а може не одна
+        if result > 25:  # Enter the action to take if the result is greater than 25
+            break
+        print(str(number) + "x" + str(multiplier) + "=" + str(result))  # Increment the appropriate variable
+        multiplier += 1
+
+
+multiplication_table(3)
+# Should print:
+# 3x1=3
+# 3x2=6
+# 3x3=9
+# 3x4=12
+# 3x5=15
+
+
+# task 2
+"""  Написати функцію, яка обчислює суму двох чисел.
+"""
+
+
+def sum_numbers(a, b):
+    return a + b
+
+
+result = sum_numbers(3, 5)
+print(result)
+
+# task 3
+"""  Написати функцію, яка розрахує середнє арифметичне списку чисел.
+"""
+
+
+def num_average(numbers):
+    if not numbers:
+        return 0
+    return sum(numbers) / len(numbers)
+
+
+num_list = [15, 7, 39, 51, 46]
+average = num_average(num_list)
+print(average)
+
+# task 4
+"""  Написати функцію, яка приймає рядок та повертає його у зворотному порядку.
+"""
+
+
+def reverse_str(s):
+    return s[::-1]
+
+
+new_string = [97, 63, 75, 25, 5]
+reversed_str = reverse_str(new_string)
+print(reversed_str)
+
+# task 5
+"""  Написати функцію, яка приймає список слів та повертає найдовше слово у списку.
+"""
+
+
+def find_longest_word(words):
+    if not words:
+        return ""
+    long_word = max(words, key=len)

```

Comment:
It is no need to create separate local variable. It is enough to return max(arg) value.

Reply:
Thank you for marks and explanations
will keep them in mind 🙂